autorest/azure/token.go

```diff
@@ -104,6 +154,79 @@ func (spt *ServicePrincipalToken) EnsureFresh() error {
 
 // Refresh obtains a fresh token for the Service Principal.
 func (spt *ServicePrincipalToken) Refresh() error {
+	newToken, err := spt.GetOauthToken()
```

Comment:
Yah! I try to balance two goals: Get what I need done with a few lines /  methods as possible and don't make any one method too long (e.g., more than, say, 50-75 lines).